src/main/js/SSEConnection.js

```diff
@@ -151,15 +151,16 @@ SSEConnection.prototype = {
                 var listenUrl = sseConnection.jenkinsUrl + '/sse-gateway/listen/'
                     + encodeURIComponent(tabClientId);
 
+                var options = undefined;
                 if (sseConnection.configuration.sendSessionId) {
                     // Sending the jsessionid helps headless clients to maintain
                     // the session with the backend.
                     var jsessionid = response.data.jsessionid;
-                    listenUrl += ';jsessionid=' + jsessionid;
+                    options = {headers: {Cookie: 'jsessionid=' + jsessionid}}
                 }
 
                 var EventSource = window.EventSource;
-                var source = new EventSource(listenUrl);
+                var source = new EventSource(listenUrl, options);
```

Comment:
Any problems with passing `undefined` here in non-test code?

Reply:
When you have a JavaScript method expecting 2 parameters where the second is optional, calling a(b) or a(b, undefined) is "similar". I tested manually this one to be sure ;)